Mathlib/Analysis/SpecialFunctions/NthRoot.lean

```diff
@@ -0,0 +1,93 @@
+import Mathlib.Analysis.SpecialFunctions.Pow.Real
+
+/-!
+# nth root operations
+
+This file provides `Real.nthRoot n` to compute `ⁿ√`,
+which operates as expected on negative values when `n` is odd.
+The trap this avoids is that using `rpow`, `(-8 : ℝ) ^ (1/3 : ℝ) = 1`.
+
+-/
+
+noncomputable section
+
+namespace Real
+
+def nthRoot (n : ℕ) (r : ℝ) : ℝ :=
```

Comment:
Do we want to allow `n` to be an `Int`? To be a `Rat`? Should we instead define this via an `instance : Pow ℝ ℚ, and have `abbrev nthRoot := r ^ (n⁻¹ : ℚ)`?

Reply:
I'm not a fan of that instance. It would have the property that `r ^ (n⁻¹ : ℚ) ≠ r ^ (n⁻¹ : ℝ)` in general.